File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/medal/MedalCreator.java
package com.depromeet.threedollar.domain.rds.user.domain.medal;

import org.jetbrains.annotations.NotNull;
import org.springframework.util.StringUtils;

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture;

import lombok.Builder;

@TestFixture
public class MedalCreator {

    @Builder
    public static Medal create(
        @NotNull String name,
        String introduction,
        String activationIconUrl,
        String disableIconUrl,
        MedalAcquisitionConditionType conditionType,
        Integer conditionCount,
        String acquisitionDescription
    ) {
        if (!StringUtils.hasText(introduction)) {
            introduction = "메달 설명";
        }
        if (!StringUtils.hasText(activationIconUrl)) {
            activationIconUrl = "iconUrl";
        }
        if (!StringUtils.hasText(disableIconUrl)) {
            disableIconUrl = "disableUrl";
        }
        if (conditionType == null) {
            conditionType = MedalAcquisitionConditionType.ADD_STORE;
        }
        if (conditionCount == null) {
            conditionCount = 3;
        }

        return Medal.builder()
            .name(name)
            .introduction(introduction)
            .activationIconUrl(activationIconUrl)
            .disableIconUrl(disableIconUrl)
            .conditionType(conditionType)
            .conditionCount(conditionCount)
            .acquisitionDescription(acquisitionDescription)
            .build();
    }

}

File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/medal/MedalCreator.kt
package com.depromeet.threedollar.domain.rds.user.domain.medal

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture

@TestFixture
object MedalCreator {

    @JvmOverloads
    @JvmStatic
    fun create(
        name: String,
        conditionType: MedalAcquisitionConditionType = MedalAcquisitionConditionType.ADD_STORE,
        conditionCount: Int = 3
    ): Medal {
        return Medal.builder()
            .name(name)
            .introduction("메달 설명")
            .activationIconUrl("iconUrl")
            .disableIconUrl("disableUrl")
            .conditionType(conditionType)
            .conditionCount(conditionCount)
            .build()
    }

    @JvmOverloads
    @JvmStatic
    fun create(
        name: String,
        introduction: String,
        activationIconUrl: String,
        disableIconUrl: String,
        conditionType: MedalAcquisitionConditionType = MedalAcquisitionConditionType.ADD_STORE,
        conditionCount: Int = 3,
        acquisitionDescription: String? = null
    ): Medal {
        return Medal.builder()
            .name(name)
            .introduction(introduction)
            .activationIconUrl(activationIconUrl)
            .disableIconUrl(disableIconUrl)
            .conditionType(conditionType)
            .conditionCount(conditionCount)
            .acquisitionDescription(acquisitionDescription)
            .build()
    }

}
